maven-core/src/test/java/org/apache/maven/internal/transformation/ConsumerPomArtifactTransformerTest.java

```diff
@@ -60,9 +56,15 @@ void transform() throws Exception {
         try (InputStream expected = Files.newInputStream(beforePomFile)) {
             Model model = new Model(new MavenStaxReader().read(expected));
             MavenProject project = new MavenProject(model);
-            ConsumerPomArtifactTransformer t = new ConsumerPomArtifactTransformer(modelBuilder);
-            t.createConsumerPomArtifact(project, tempFile, systemSessionMock)
-                    .transform(beforePomFile, tempFile, model.getDelegate());
+            ConsumerPomArtifactTransformer t = new ConsumerPomArtifactTransformer((s, p, f) -> {
+                try (InputStream is = Files.newInputStream(f)) {
+                    return DefaultConsumerPomBuilder.transform(new MavenStaxReader().read(is));
+                } catch (Exception e) {
```

Comment:
not needed, the exception fails the test

Reply:
If you mean "you should not catch the exception and let it bubble through and let the exception make the test fail", it unfortunately is not doable because of the usage of checked exceptions.  Adding an exception to the interface would go up the chain to https://github.com/apache/maven/blob/7b38457506544854cdd26aa1d49608050217b923/maven-artifact/src/main/java/org/apache/maven/artifact/Artifact.java#L82 which is part of the API and should not be changed.  So while I would happily remove it if possible, I don't see any good solution.